Kernel_23/include/CGAL/Bbox.h

```diff
@@ -0,0 +1,161 @@
+
+#ifndef CGAL_BBOX_H
+#define CGAL_BBOX_H
+
+#include <boost/config.hpp> // defines BOOST_PREVENT_MACRO_SUBSTITUTION
+#include <stddef.h>
+#include <limits>
+#include <array>
+#include <iostream>
+
+namespace CGAL {
+namespace Impl {
+
+template<typename Container, typename Derived>
+class Bbox
+{
+protected:
+    typedef typename Container::value_type T;
+    Container min_values;
+    Container max_values;
+
+public:
+    Bbox& operator+=(const Bbox& bbox)
+    {
+        CGAL_assertion(min_values.size() == 0 || min_values.size() == bbox.min_values.size());
+        if(min_values.size() == 0){
+            *this =  bbox;
+        }
+        int dim = bbox.min_values.size();
+        for(int i=0; i<dim; ++i)
+        {
+            if(min_values[i] > bbox.min_values[i]) min_values[i] = bbox.min_values[i];
+            if(max_values[i] < bbox.max_values[i]) max_values[i] = bbox.max_values[i];
+        }
+        return *this;
+    }
+
+    inline int dimension() const
+    {
+        return static_cast<const Derived*>(this)->dimension();
+    }
+
+    inline T min BOOST_PREVENT_MACRO_SUBSTITUTION (int i) const
+    {
+        return min_values[i];
+    }
+
+    inline T max BOOST_PREVENT_MACRO_SUBSTITUTION (int i) const
+    {
+        return max_values[i];
+    }
+
+    inline T& min BOOST_PREVENT_MACRO_SUBSTITUTION (int i)
+    {
+        return min_values[i];
+    }
+
+    inline T& max BOOST_PREVENT_MACRO_SUBSTITUTION (int i)
+    {
+        return max_values[i];
+    }
+
+    inline T measure() const {
+        T result = max_values[0] - min_values[0];
+        if (result <= 0) return 0;
+        for(int i=1; i<dimension(); ++i)
+            result *= max_values[i] - min_values[i];
+        return result;
+    }
+
+    inline T intersection_measure(const Bbox& bbox) const {
+        CGAL_assertion(dimension() == bbox.dimension());
+        T result = 1;
+        for(int i=0; i<dimension(); ++i) {
+            result *= std::min(max(i), bbox.max(i)) -
+                      std::max(min(i), bbox.min(i));
+            if (result <= 0) return 0;
+        }
+        return result;
+    }
+
+    bool operator==(const Bbox& bbox) const {
+        for(int i=0; i<dimension(); ++i)
+            if (min_values[i] != bbox.min_values[i] || max_values[i] != bbox.max_values[i])
+                return false;
+        return true;
+    }
+
+    bool operator!=(const Bbox& bbox) const { return !operator==(bbox); }
+
+protected:
+    void init(int d, T range = -std::numeric_limits<T>::infinity()) {
+        for(int i=0; i<d; ++i)
+        {
+            min_values[i] = -range;
+            max_values[i] =  range;
+        }
+    }
+};
+
+}
+
+// A D-dimensional axis aligned box
+template<unsigned int N, typename T>
+class Bbox : public Impl::Bbox<std::array<T, N>, Bbox<N,T>>
+{
+    enum { D = N };
+public:
+    inline constexpr int dimension() const { return D; }
+    Bbox(int d = 0          ) { assert(d==N || d==0); this->init(d       ); }
```

Comment:
I find it surprising that for a Bbox with non-dynamic dimension I have to pass parameter `d`.

Reply:
that was to offer a common API whatever the dimension dynamics